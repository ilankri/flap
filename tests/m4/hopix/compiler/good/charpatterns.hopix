fun f (x : char) : int =
  x ? { 'a' => 1 | 'b' => 0 | (_ : char) => -1 }

fun check (x : int, expected : int) : unit =
  print_int (x);
  if x = expected then print_string (" OK\n") else print_string (" KO\n")

val test =
  check (f ('a'), 1);
  check (f ('b'), 0);
  check (f ('z'), -1)