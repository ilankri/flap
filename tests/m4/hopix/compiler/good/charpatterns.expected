1 OK
0 OK
-1 OK
